src/tests/Interop/COM/ComWrappers/WeakReference/WeakReferenceTest.cs

```diff
@@ -132,6 +134,7 @@ private static void ValidateWeakReferenceState(WeakReference<WeakReferenceableWr
                 Assert.Equal(sourceWrappers.Registration, target.Registration);
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
This makes sense, but I'm going to ask you to make a slightly more complicate fix. Can you please convert both of this APIs into `static` local functions into the `ValidateNativeWeakReference` test? Setting the `[MethodImpl(MethodImplOptions.NoInlining)]` is also appropriate. We should also update the other `GetWeakReference` local functions that are in this file to be `static` with the same impl attribute.

Reply:
> Can you please convert both of this APIs into static local functions into the ValidateNativeWeakReference test?

I was actually wondering why we do not do the opposite. 

Local functions reduce readability. That is the reason why I missed other local functions in this file - because it is hard to see that there are helper functions when they are hidden inside other functions and look like method blocks. 

I think for a testcase it is more important that the code is simple and readable.

Reply:
Actually, once there are attributes, it may be easier to spot them.

Reply:
> Local functions reduce readability.

A rather subjective opinion that I empathize with. The most important consideration for me is being consistent so if you want to convert them to `static` methods on the class that is also okay with me.

Reply:
I will also move them to the top of containing methods - so that they are not mixed up with sequential code.

Reply:
Local functions have benefits. The primary one is encapsulation and not polluting the outer name space. I just wish the syntax would make them stand out more, or perhaps visualize them in VisualStudio somehow - highlight the background a little, or something.

Anyways. I've updated other cases of the same pattern.